daisy/step_wait_for_instances_signal.go

```diff
@@ -33,13 +33,14 @@ type WaitForInstancesSignal []*InstanceSignal
 // port.
 // A StatusMatch will print out the matching line from the StatusMatch onward.
 // This step will not complete until a line in the serial output matches
-// SuccessMatch or FailureMatch. A match with FailureMatch will cause the step
-// to fail.
+// SuccessMatch, FailureMatch or FailureMatches. A match with FailureMatch or FailureMatches will
+// cause the step to fail.
 type SerialOutput struct {
-	Port         int64  `json:",omitempty"`
-	SuccessMatch string `json:",omitempty"`
-	FailureMatch string `json:",omitempty"`
-	StatusMatch  string `json:",omitempty"`
+	Port           int64    `json:",omitempty"`
+	SuccessMatch   string   `json:",omitempty"`
+	FailureMatch   string   `json:",omitempty"`
+	FailureMatches []string `json:",omitempty"`
```

Comment:
If you write a custom unmarshaler you can have the existing FailureMatch take []string, see Var as an example: https://github.com/GoogleCloudPlatform/compute-image-tools/blob/master/daisy/workflow.go#L77

Reply:
This is now deprecated after the refactoring to support both arrays and single strings in the old (FailureMatch) field.